src/helpers/gateway/mod.rs

```diff
@@ -59,6 +60,34 @@ impl<T: Transport> Gateway<T> {
         roles: RoleAssignment,
         transport: T,
     ) -> Self {
+        let (tx, rx) = mpsc::channel();
+        let senders = Arc::new(Mutex::new(GatewaySenders::default()));
+        let senders_clone = Arc::clone(&senders);
+        let receivers = Arc::new(Mutex::new( GatewayReceivers::default()));
+        let receivers_clone = Arc::clone(&receivers);
+        thread::spawn(move|| {
+            // Perform some periodic work in the background
+            loop {
+                thread::sleep(Duration::from_secs(1));
+                let senders = senders_clone.lock().unwrap();
+                let receivers = receivers_clone.lock().unwrap();
+                if senders.is_idle() && receivers.is_idle(){
+                    //TODO: print something
+                    println!("{:?}: idle Pending messages: {:?}", thread::current().id(), senders.get_all_pending_records());
+                }
+                else {
+                    senders.reset_idle();
+                    receivers.reset_idle();
+                }
+                match rx.try_recv()
```

Comment:
we should just keep the handle returned by spawn method inside the gateway. In this case we don't need a sender to terminate it, we just call abort inside `Drop`

Reply:
sounds good